<launch>
    <arg name="use_robot"       default="true"/>
    <arg name="camera_topic"    default="/misty/id_0/camera/color/image_raw"/>
    <arg name="use_s2t"         default="false"/>
    <arg name="use_tts"         default="false"/>
    <arg name="audio_device"    default="7"/> 

    <group if="$(arg use_robot)">
        <include file="$(find misty_wrapper)/launch/misty.launch" pass_all_args="true"></include>
    </group>
    
    <node pkg="gaze_tracking_ros" type="gaze_tracker.py" name="gaze_tracker">
        <remap from="image_raw" to="$(arg camera_topic)" />
    </node>

    <node name="gaze_follower" pkg="robo_copilot" type="gaze_follower.py" output="screen"/>

    <group if="$(arg use_s2t)">
        <include file="$(find ros_speech2text)/launch/ros_speech2text.launch">
            <arg name="audio_device_idx" value="$(arg audio_device)"/>
        </include>
    </group>

    <group if="$(arg use_tts)">
        <node name="google_tts" pkg="robo_copilot" type="google_tts.py" output="screen"/>
    </group>

    <node name="cpp_editor_node" pkg="robo_copilot" type="tk_editor.py" output="screen"/>
    <node name="feedback_gen" pkg="robo_copilot" type="feedback_gen.py" output="screen"/>

    <!-- <node name="code_listener" pkg="robo_copilot" type="code_listener.py" output="screen"/> -->

</launch>